src/modules/Elsa.MassTransit.RabbitMq/Features/RabbitMqServiceBusFeature.cs

```diff
@@ -32,18 +37,40 @@ public RabbitMqServiceBusFeature(IModule module) : base(module)
     /// <inheritdoc />
     public override void Configure()
     {
-        Module.Configure<MassTransitFeature>().BusConfigurator = configure =>
+        Module.Configure<MassTransitFeature>(massTransitFeature =>
         {
-            configure.UsingRabbitMq((context, serviceBus) =>
+            massTransitFeature.BusConfigurator = configure =>
             {
-                if (!string.IsNullOrEmpty(ConnectionString))
-                    serviceBus.Host(ConnectionString);
+                var tempConsumers = massTransitFeature.GetConsumers()
+                    .Where(c => c.IsShortLived)
+                    .ToList();
+                
+                configure.AddConsumers(tempConsumers.Select(c => c.ConsumerType).ToArray());
+                
+                configure.UsingRabbitMq((context, serviceBus) =>
+                {
+                    var options = context.GetRequiredService<IOptions<MassTransitWorkflowDispatcherOptions>>().Value;
+                    var instanceNameRetriever = context.GetRequiredService<IInstanceNameRetriever>();
+                    
+                    if (!string.IsNullOrEmpty(ConnectionString))
+                        serviceBus.Host(ConnectionString);
 
-                ConfigureServiceBus?.Invoke(serviceBus);
+                    ConfigureServiceBus?.Invoke(serviceBus);
+                    
+                    foreach (var consumer in tempConsumers)
+                    {
+                        serviceBus.ReceiveEndpoint($"{instanceNameRetriever.GetName()}-{consumer.Name}", configurator =>
+                        {
+                            configurator.QueueExpiration = options.ShortTermQueueLifetime ?? TimeSpan.FromHours(1);
+                            configurator.ConcurrentMessageLimit = options.ConcurrentMessageLimit;
+                            configurator.ConfigureConsumer<DispatchCancelWorkflowsRequestConsumer>(context);
+                        });
+                    }
 
-                serviceBus.ConfigureEndpoints(context, new KebabCaseEndpointNameFormatter("Elsa", false));
-            });
-        };
+                    serviceBus.ConfigureEndpoints(context, new KebabCaseEndpointNameFormatter("Elsa", false));
```

Comment:
Some of these blocks might be moved to an extension - both RabbitMQ and Azure Service Bus configurators share a common ancestor. I did [something similar in the dispatch-channels branch](https://github.com/elsa-workflows/elsa-core/blob/feature/dispatch-channels/src/modules/Elsa.MassTransit/Extensions/ServiceBusChannelsConfigurationExtensions.cs) - that extension is used in both RabbitMQ and ASB configuration for MT.